.ci/jenkins/runCI.sh

```diff
@@ -16,142 +16,281 @@
 # limitations under the License.
 # ******************************************************************************
 
-CI_PATH="$( cd "$(dirname "$0")" ; pwd -P )"
-CI_ROOT=".ci/jenkins"
-REPO_ROOT="${CI_PATH%$CI_ROOT}"
-DOCKER_CONTAINER="ngraph-onnx_ci_repro"
-
-function clone_ngraph() {
-    local sha="$1"
-
-    set -x
-    cd "${REPO_ROOT}"
-    if [ -d "./ngraph" ]; then
-        echo "[WARNING!] nGraph repo found! Skip cloning!"
-    else
-        git clone https://github.com/NervanaSystems/ngraph
+HELP_MESSAGE="Usage: runCI.sh [ARGS] [--help]
+
+Builds nGraph and runs nGraph-ONNX tests.
+
+Arguments:
+    --help                      - Displays this message.
+    --cleanup                   - Removes docker container and files created during script execution, instead of running CI.
+    --rebuild                   - Rebuilds nGraph and runs tox tests.
+    --ngraph-branch=...         - Will use specified branch of nGraph.
+                                  Default: master
+    --ngraph-sha=...            - Checkout to specified nGraph commit.
+                                  Default: none - latest commit of cloned branch used
+    --backends=...              - Comma separated list (no whitespaces!) of nGraph backends to run CI on.
+                                  Default: cpu,interpreter
+"
+
+NGRAPH_REPO_ADDRESS="https://github.com/NervanaSystems/ngraph.git"
+NGRAPH_REPO_DIR_NAME="ngraph"
+
+DEFAULT_NGRAPH_REPO_BRANCH="master"
+DEFAULT_NGRAPH_REPO_SHA=""
+
+# "<OPERATING_SYSTEM>" will be replaced by particular OS during script execution
+DOCKER_CONTAINER_NAME_PATTERN="ngraph-onnx_ci_reproduction_<OPERATING_SYSTEM>"
+DOCKER_IMAGE_NAME_PATTERN="aibt/aibt/ngraph/<OPERATING_SYSTEM>/base"
+DOCKER_BASE_IMAGE_TAG="ci"
+DOCKER_EXEC_IMAGE_TAG="ci_run"
+DOCKER_HOME="/home/${USER}"
+
+DEFAULT_BACKENDS="cpu interpreter"
+HTTP_PROXY="${http_proxy}"
+HTTPS_PROXY="${https_proxy}"
+
+function main() {
+    # Main function
+
+    # Load parameters defaults
+    CLEANUP="false"
+    REBUILD="false"
+    NGRAPH_REPO_BRANCH="${DEFAULT_NGRAPH_REPO_BRANCH}"
+    NGRAPH_REPO_SHA="${DEFAULT_NGRAPH_REPO_SHA}"
+    BACKENDS="${DEFAULT_BACKENDS}"
+
+    parse_arguments "${@}"
+
+    NGRAPH_ONNX_CI_ABS_PATH="$(pwd)/$( dirname "${BASH_SOURCE[0]}" )"
+    NGRAPH_ONNX_ROOT_ABS_PATH="${NGRAPH_ONNX_CI_ABS_PATH%/.ci*}"
+    local ngraph_onnx_parent_path="$(dirname ${NGRAPH_ONNX_ROOT_ABS_PATH})"
+    WORKSPACE="${ngraph_onnx_parent_path}"
+    NGRAPH_REPO_PATH="${WORKSPACE}/${NGRAPH_REPO_DIR_NAME}"
```

Comment:
In my local environment it refers to my home directory (after path operations) and during building context to Docker daemon (Building base image) it takes a planty of GB (all home dir)

Reply:
Fixedin latest commit